src/ctap/mod.rs

```diff
@@ -919,6 +923,63 @@ where
         Ok(ResponseData::AuthenticatorSelection)
     }
 
+    fn process_vendor_configure(
+        &mut self,
+        params: AuthenticatorVendorConfigureParameters,
+        cid: ChannelID,
+    ) -> Result<ResponseData, Ctap2StatusCode> {
+        (self.check_user_presence)(cid)?;
+
+        // Sanity checks
+        let has_priv_key = self.persistent_store.attestation_private_key()?.is_some();
+        let has_cert = self.persistent_store.attestation_certificate()?.is_some();
+
+        if params.attestation_material.is_some() {
+            let data = params.attestation_material.unwrap();
+            if !has_cert {
```

Comment:
I thought about it. But what if, for example, writing the certificate works but fails for the private key?
I saw 3 solutions:
1. modify the cbor command to allow setting certificate and private key individually or together
1. accept but ignore an already set value which allows to recover from the described scenario
1. delete the certificate if the private key fails.

Reply:
I see. Both 2 and 3 look good (2 being the current code). For 2 we could additionally check that the ignored value is equal to the existing one. But I'm fine with the current code, was just asking.

Reply:
Why is overwriting so bad on a non-locked-down device? Maybe just make writing conditional, and since lockdown depends on both being set, you should be good?

Reply:
I think the code deals equally well with none of the 2 being set, or only 1 being set.

Reply:
The storage doesn't know if the device is locked or not. And that's the one preventing any overwrite :)


> For 2 we could additionally check that the ignored value is equal to the existing one

Good point, I'll add that extra check!

Reply:
So solution 4 would be to have a `is_locked_down` in persistent storage, that reflects that. And make writing to the certificate conditional on that. It would also allow to return the state of `is_locked_down` as part of this command as you planned.

Reply:
Doing this means that we have one value stored at 2 places which can go out of sync.

Reply:
From another thread I understood that we can't know if the device is locked, so we don't really have it in 2 places. If it's possible to fetch the value from some register (maybe adding a syscall to the new driver, e.g. command 0 could return whether the device is protected), then that would be the best solution no?

Reply:
I updated HIL + capsule + syscalls to have a get/set model and support different levels of protection. For the vendor specific command I keep a boolean because it doesn't make sense in our context to partially protect.

Regarding our implementation and allowing overwriting the certificate and the private key on a non-locked device, I still see an issue allowing it: it would invalidate all credentials that have been created using U2F. This would only make sense in combination with a reset command (although I'm not familiar enough with batch attestation and the implications it may have). With CTAP2.1 though, some credentials may persist even after a reset and could still risk to be invalidated.

Reply:
I think the current implementation is solid, then.